CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+work with the files in the `dev/` directory than on a phone or emulator. Below is
+an explanation of the files and their purpose.
+
+| File | Purpose |
+|----------|-----|
+| `dev.html` | Open this page in a browser after running `$ npm run dev`. |
+| `dev.js` | Injects the components and dependencies into the window the same way `index.js` would. |
+| `config.js` | Clay config to use as a sandbox for testing components. |
+| `custom-fn.js` | Clay custom function to be injected by `dev.js`. |
+| `emulator.html` | Copy of the HTML page that is used to make the Clay compatible with the Pebble SDK emulator. |
+| `uri-test.html` | Used to stress test URI creation for older browser versions. |
+
+
+## Testing your change inside of a test project.
+
+Once your change is ready to be tested in the emulator or on the watch, follow these steps:
+
+1. From your Clay fork run `$ npm run pebble-build`. This builds Clay to work with the Pebble SDK
+2. From your test Pebble project, run `$ pebble package install /path/to/your/clay/fork`.
+  This will modify your `package.json` to point directly to your local copy of Clay.
+3. Run the usual`$ pebble build && pebble install --emulator basalt && pebble emu-app-config` to test your modifications
+4. If you make modifications to your fork of Clay, you must repeat the above steps.
+
+
+## Functionality
+
+Most of the magic happens in the `src/scripts/lib` directory. `config-page.js`
+initializes a new instance of `ClayConfig` and calls the injected custom function
+(`window.customFn`) with the `ClayConfig` as its context. This allows developers to
+add extra functionality to the config page, such as setting values of items dynamically
+or registering small custom components.
+
+Once the `ClayConfig` is initialized, we run the `.build()` method. This iterates over
+the config and injects each item into the page. Each item is an instance of `ClayItem`.
+It also indexes the items to later be retrieved with `.getAllItems()`,
+`.getItemByMessageKey()`, `.getItemById()`, `.getItemsByType()`.
+
+
+## Testing
+
+Clay enforces 100% test coverage. Chances are, that if you are submitting a pull request,
+there should be tests for the proposed code changes. Have a look at the other tests in
+the repository to see some examples of how you should be writing your tests.
+
+There are two commands to use while writing your tests.
+
+ - `$ npm run test` - Runs all tests once.
+ - `$ npm run test-debug` - Watches for file changes and re-runs all tests when it
+  detects a change. This also leaves an in instance of Google Chrome running.
+  If you click the "Debug" button in the top right of the page, a new tab will open.
+  You can then use the Google Chrome developer tools to debug your tests as they run.
+  Refreshing the page will re-run the tests.
+
+## Style Guide
+
+Have a look at the rest of the codebase for an example of how you should format your code.
```

Comment:
I'm not sure I would describe "look at the rest of our codebase for an example" as being a style guide..